src/fparser/two/tests/utils/test_blockbase.py

```diff
@@ -87,3 +88,96 @@ def test_include(f2003_create):
         "'! comment5')), End_Program_Stmt('PROGRAM', Name('test'))],)") \
         in str(result)
     assert "should" not in str(result)
+
+
+@pytest.mark.parametrize('strict_order', [True, False])
+def test_strict_order_invalid_code(f2003_create, strict_order):
+    '''Check that the strict_order flag toggles the parse behaviour as
+    expected.
+    '''
+    subclasses = [F2003.Specification_Part, F2003.Execution_Part]
+    reader = get_reader("""
+        program main
+            i = 2
+            integer :: i
+        end program main
+        """)
+
+    expected = remove_indentation("""([
+      Program_Stmt('PROGRAM', Name('main')),
+      Execution_Part(
+          Assignment_Stmt(Name('i'), '=',
+          Int_Literal_Constant('2', None))),
+      Specification_Part(
+          Type_Declaration_Stmt(
+              Intrinsic_Type_Spec('INTEGER', None),
+              None,
+              Entity_Decl(Name('i'), None, None, None))),
+      End_Program_Stmt('PROGRAM', Name('main'))
+     ],)
+    """)
+    result = BlockBase.match(
+        F2003.Program_Stmt, subclasses, F2003.End_Program_Stmt, reader,
+        strict_order=strict_order)
+
+    if strict_order:
+        assert result == None
```

Comment:
pycodestyle and pylint say to use 'result is None'

Reply:
Aha. Interesting, I thought pytest needed the equality to do its magic (and provide a good message) but it turns out it doesn't. :+1: